cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -217,6 +222,7 @@ public Module getDistributedModules() {
                              protected void configure() {
                                bind(SchedulerService.class).to(DistributedSchedulerService.class).in(Scopes.SINGLETON);
                                bind(Scheduler.class).to(SchedulerService.class);
+                               bind(MRJobInfoFetcher.class).to(MRJobClientWithMetricsFallback.class);
```

Comment:
I figured the name should be more closely related to its implementation.
So `Metrics` in local and `JobClient` in distributed.
Is that not a good idea?